aQute.libg/src/aQute/lib/filter/Filter.java

```diff
@@ -33,7 +40,20 @@
 		}
 	}
 
-	final char			WILDCARD	= 65535;
+	final static int				cacheSize	= 10000;
+
+	@SuppressWarnings({
+		"rawtypes", "serial", "unchecked"
+	})
+	final static Map<String, Node>	cache		= Collections
+		.synchronizedMap(new LinkedHashMap<String, Node>(128, 0.75f, true) {
```

Comment:
Funny story, I started with a WeakHashMap. The problem is that there's so much garbage produced (and thus GC) by the verification process that the task actually took 50x longer :|

Reply:
perhaps, what would that look like?

Reply:
Well it might require a large restructuring of this class to separate parsing from matching. Right now the code does the parsing on the match call which seems wrong anyway since you might call match multiple time on a filter expression. This change alone may be a perf win without caching parsed filters.